src/Api/DataProvider/AbstractElasticSearchDataProvider.php

```diff
@@ -58,8 +60,10 @@ abstract class AbstractElasticSearchDataProvider
      *   Service for registering no hits
      * @param loggerInterface $logger
      *   Standard logger
+     * @param ParameterBagInterface $params
+     *   Access to environment variables
      */
-    public function __construct(Type $index, RequestStack $requestStack, StatsLoggingService $statsLoggingService, MetricsService $metricsService, EventDispatcherInterface $dispatcher, IdentifierFactory $factory, NoHitService $noHitService, LoggerInterface $logger)
+    public function __construct(Type $index, RequestStack $requestStack, StatsLoggingService $statsLoggingService, MetricsService $metricsService, EventDispatcherInterface $dispatcher, IdentifierFactory $factory, NoHitService $noHitService, LoggerInterface $logger, ParameterBagInterface $params)
```

Comment:
It is better to inject the variable than the parameter bag.

Reply:
I have just followed the pattern used in the rest of cover-service and cover-service-importers!